genesis/sensors/base_sensor.py

```diff
@@ -162,31 +174,104 @@ def read_ground_truth(self, envs_idx: List[int] | None = None):
         """
         return self._get_formatted_data(self._manager.get_cloned_from_cache(self, is_ground_truth=True), envs_idx)
 
+    def add_recorder(self, handler: "DataHandler", rec_options: RecordingOptions | None = None):
+        """
+        Add a sensor data recorder that processes data from this sensor.
+        """
+        if self._data_recorder is None:
+            self._data_recorder = self._manager.create_data_recorder(self)
+        self._data_recorder.add_handler(handler, rec_options)
+
+    def start_recording(self):
```

Comment:
Same. You should explain what "recording" means in this context. It is far from obvious for an outsider. "recording" is not a standard terminology of this. Usually, it is more about "robot telemetry". I'm fine with recording, but you need to explain what it does AND what can be done with the resulting file, i.e. what is the format internally, what is the right way to load them, and eventually display them.